CMakeLists.txt

```diff
@@ -314,6 +318,25 @@ option(NO_DX "Disable DirectX support")
 option(NO_TESTS "Disable all tests")
 option(NO_GLTESTS "Disable GL tests")
 
+option(MSVC_STATIC_CRT_RELEASE "Statically link MSVC CRT in Release" OFF)
+option(MSVC_STATIC_CRT_DEBUG "Statically link MSVC CRT in Debug" OFF)
+
+if(MSVC AND MSVC_STATIC_CRT_RELEASE)
+    message(STATUS "Using static MSVC CRT Release")
+    # http://stackoverflow.com/a/32128977/486990
+    add_compile_options(
+        "$<$<CONFIG:RelWithDebInfo>:/MT>"
+        "$<$<CONFIG:Release>:/MT>"
+        "$<$<CONFIG:MinSizeRel>:/MT>"
+    )
+endif()
+if(MSVC AND MSVC_STATIC_CRT_DEBUG)
```

Comment:
I see your point, if you're building a single variant into your project, and the build files for OSD are local to your project that makes total sense.

The four variants are Static-Debug, Static-Release, Dynamic-Debug, Dynamic-Release. I don't think we have to worry about MultiThreaded or non-MT any more (so thankfully there aren't 8 variations).

In a larger build ecology it's common to build all the variants for packaging so that individual programmers don't have to do the builds at their desks. Of course one could re-run cmake to generate all the variants and do builds but that's far less convenient than having all the variants in a single project file because one only has to run msbuild four times.